### PR TITLE
BaSP-MR-51: Member

### PR DESCRIPTION
**Pull request by OscarLucero**
Created Methods:
-Update
-Delete
-Create

### CODE REVIEW

File: src/Components/Members/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import React from 'react';
```
Comment (JuanManuel-Blois) [inline src/Components/Members/Form/index.jsx]: We were told to put everything in the same return, from and table should be in the Members index

File: src/Components/Members/Table/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (JuanManuel-Blois) [inline src/Components/Members/Table/index.jsx]: We were told to put everything in the same return, from and table should be in the Members index

File: src/Components/Members/index.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    phone: '',
+    city: '',
+    birthday: '2023-05-29T07:32:26+0000',
+    postalCode: '',
+    membership: '',
+    isActive: true
+  });
+  const [idMember, setIdMember] = useState('');
+  const getMembers = async () => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/members`);
```
Comment (JuanManuel-Blois) [inline src/Components/Members/index.jsx]: remember to implement the /api/members fix discussed on slack
Screenshot 2023-05-30 at 23 26 32
fix Image 2023-05-30 at 20 39 22

File: src/Components/Members/index.jsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const addMember = async (member) => {
+    const dateFormat = changeDateFormat(member.birthday);
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/members`, {
```
Comment (JuanManuel-Blois) [inline src/Components/Members/index.jsx]: remember to implement the /api/members fix discussed on slack
Screenshot 2023-05-30 at 23 26 32
fix Image 2023-05-30 at 20 39 22

File: src/Components/Members/index.jsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        setMembers([...members, member]);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const deleteMember = async (memberId) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/members/${memberId}`, {
```
Comment (JuanManuel-Blois) [inline src/Components/Members/index.jsx]: remember to implement the /api/members fix discussed on slack
Screenshot 2023-05-30 at 23 26 32
fix Image 2023-05-30 at 20 39 22

File: src/Components/Members/index.jsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    let day = dateArray[2];
+
+    dateArray = month + '/' + day + '/' + year;
+    return dateArray;
+  };
+
+  const updateMember = async (idMember) => {
+    const dateFormat = changeDateFormat(memberValues.birthday);
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/members/${idMember}`, {
```
Comment (JuanManuel-Blois) [inline src/Components/Members/index.jsx]: remember to implement the /api/members fix discussed on slack
Screenshot 2023-05-30 at 23 26 32
fix Image 2023-05-30 at 20 39 22

**Review by leimeter-joaquin — COMMENTED** [review]
Check the comments made by Juanma, everything should go in one component as stated in the planning.

File: src/Components/Members/index.jsx [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+
+      const updatedMember = await response.json();
+      if (response.ok) {
+        const dataIndex = members.findIndex((Member) => Member._id === idMember);
+
+        setMembers((currentMembers) => {
+          const updatedMembers = [...currentMembers];
+          updatedMembers[dataIndex] = updatedMember.data;
+          return updatedMembers;
+        });
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx:99]: you should send success mesages or fail messages

File: src/Components/Members/index.jsx [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ ...member, birthday: dateFormat })
+      });
+      if (response.ok) {
+        setMembers([...members, member]);
+      }
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx:127]: same here, a message would be fine  to know which fields are incorrect.

File: src/Components/Members/index.jsx [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+    }
+  };
+
+  const deleteMember = async (memberId) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${memberId}`, {
+        method: 'DELETE'
+      });
+      console.log(response);
+      setMembers([...members.filter((member) => member._id !== memberId)]);
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx:139]: here too

File: src/Components/Members/index.jsx [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+                  type="number"
+                  value={memberValues.postalCode}
+                  onChange={(e) => onChange(e)}
+                />
+              </div>
+            </div>
+            <div>
+              <div className={styles.inputContainer}>
+                <label className={styles.label}>Membership</label>
+                <input
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx]: maybe you could tell us what types of memberships are available at the gym

File: src/Components/Members/index.jsx [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+      firstName: '',
+      lastName: '',
+      email: '',
+      dni: '',
+      phone: '',
+      city: '',
+      birthday: '2023-05-29T07:32:26+0000',
+      postalCode: '',
+      membership: '',
+      isActive: true
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx:77]: you should be able to edit isActive to true or false

File: src/Components/Members/index.jsx [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+    postalCode: '',
+    membership: '',
+    isActive: true
+  });
+  const [idMember, setIdMember] = useState('');
+  const getMembers = async () => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members`);
+    const data = await response.json();
+    setMembers(data.data);
+    console.log(data);
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx]: do not forget to delete the console.log

File: src/Components/Members/index.jsx [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+        const dataIndex = members.findIndex((Member) => Member._id === idMember);
+
+        setMembers((currentMembers) => {
+          const updatedMembers = [...currentMembers];
+          updatedMembers[dataIndex] = updatedMember.data;
+          return updatedMembers;
+        });
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx]: an alert or a message in ux is preferable

File: src/Components/Members/index.jsx [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ ...member, birthday: dateFormat })
+      });
+      if (response.ok) {
+        setMembers([...members, member]);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx]: an alert or a message in ux is preferable

File: src/Components/Members/index.jsx [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+    }
+  };
+
+  const deleteMember = async (memberId) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${memberId}`, {
+        method: 'DELETE'
+      });
+      console.log(response);
```
Comment (Eugeniagassman) [inline src/Components/Members/index.jsx]: same here( console.log)

**Review by Eugeniagassman — APPROVED** [review]

**Comment by Katairala** [comment]
Tested locally on https://joaco-megarocket-app.vercel.app/
Test Results: Passed ✨ 

Validation errors will be reviewed in next instances. 

Great work.

**Pull request merged**